its/src/test/java/com/sonar/it/csharp/AnalysisWarningsTest.java

```diff
@@ -60,4 +58,13 @@ public void analysisWarningsImport() throws IOException {
     assertThat(task.getStatus()).isEqualTo(Ce.TaskStatus.SUCCESS);
     assertThat(task.getWarningsList()).containsExactly("First message", "Second message");
   }
+
+  @Test
+  public void analysisWarnings_OldRoslyn() throws IOException {
+    BuildResult buildResult = Tests.analyzeProject(temp, "Roslyn.1.3.1", null);
```

Comment:
shouldn't we actually have integration tests for MSB14 and MSB15? 

I guess we only have tests with different MSB versions on sonar-security...

Reply:
MsBuild 14 is bound with Roslyn 1.x
MsBuild 15 is bound with Roslyn 2.x
So while we could do the integration testing like that, it's not worth the effort to build the infra for it. 

Effectively, once we bump the Roslyn version, users on the old Roslyn will have trouble. And it's a good thing that all of them will already have the message.

Comments:
* We can't determine MsBuild version from within the analyzer. S4NET would have to do that. Yet not all users on SQ 10.0 will be forced to use the latest scanner.
* To align the message and the code, we should warn them about using old Roslyn. The problem with that is that users have no idea what Roslyn they use and how they are related to the build tools versions. So they wouldn't understand the problem.
* Combination of MsBuild 17 and DoetNetCompilerPlatform v1.x and v2.x package (or any other custom build solutions enforcing old Roslyn) will produce a message that is confusing for users. Yet valid to show, as they will have braking changes once we bump the version.

Reply:
did you test manually with MSB 15?

because you say that MSB15 is bound to Roslyn 2.x, but the IT is using Roslyn 1.x

or am I missing smth?

Reply:
I cannot create IT for it, because a higher version of the package is giving Roslyn 4.4.0 😒 

I plan to validate on Peach, I don't have MsBuild 15 locally.

Reference: https://learn.microsoft.com/en-us/visualstudio/extensibility/roslyn-version-support?view=vs-2022

Reply:
it should be easy to install the build tools (there's no need for a full VS instance) https://community.chocolatey.org/packages/visualstudio2017buildtools